src/main/java/org/jabref/gui/SaveOrderConfigDisplay.java

```diff
@@ -38,56 +46,87 @@ private void init() {
         fieldNames.add(BibEntry.KEY_FIELD);
         Collections.sort(fieldNames);
         String[] allPlusKey = fieldNames.toArray(new String[fieldNames.size()]);
-        savePriSort = new JComboBox<>(allPlusKey);
+        savePriSort = new ComboBox<>(FXCollections.observableArrayList(allPlusKey));
         savePriSort.setEditable(true);
-        saveSecSort = new JComboBox<>(allPlusKey);
+        saveSecSort = new ComboBox<>(FXCollections.observableArrayList(allPlusKey));
         saveSecSort.setEditable(true);
-        saveTerSort = new JComboBox<>(allPlusKey);
+        saveTerSort = new ComboBox<>(FXCollections.observableArrayList(allPlusKey));
         saveTerSort.setEditable(true);
 
-        savePriDesc = new JCheckBox(Localization.lang("Descending"));
-        saveSecDesc = new JCheckBox(Localization.lang("Descending"));
-        saveTerDesc = new JCheckBox(Localization.lang("Descending"));
-
+        savePriDesc = new CheckBox(Localization.lang("Descending"));
+        saveSecDesc = new CheckBox(Localization.lang("Descending"));
+        saveTerDesc = new CheckBox(Localization.lang("Descending"));
+
+        Font font = new Font(10);
+        GridPane builder = new GridPane();
+        Label label = new Label(Localization.lang("Primary sort criterion"));
+        label.setFont(font);
+        builder.add(label,1,1);
+        builder.add(savePriSort,2,1);
+        builder.add(savePriDesc,3,1);
+
+        Label label1 = new Label(Localization.lang("Secondary sort criterion"));
+        label1.setFont(font);
+        builder.add(label1,1,2);
+        builder.add(saveSecSort,2,2);
+        builder.add(saveSecDesc,3,2);
+
+        Label label2 = new Label(Localization.lang("Tertiary sort criterion"));
+        label2.setFont(font);
+        builder.add(label2,1, 3);
+        builder.add(saveTerSort,2,3);
+        builder.add(saveTerDesc,3,3);
+        panel = builder;
+
+        JComboBox<String> savePriSort1 = new JComboBox<>(allPlusKey);
+        savePriSort1.setEditable(true);
+        JComboBox<String> saveSecSort1 = new JComboBox<>(allPlusKey);
```

Comment:
This part here looks like a unnecessary code duplication.

Reply:
Ok, in this case you can use a `JFXPanel` in `DatabasePropertiesDialog` to display the JavaFX controls.